vmm/src/cpu.rs

```diff
@@ -329,7 +332,7 @@ impl Vcpu {
 
 pub struct CpuManager {
     boot_vcpus: u8,
-    io_bus: Arc<devices::Bus>,
+    io_bus: Weak<devices::Bus>,
```

Comment:
Why did we decide to put the communication channel between guest and VMM through PIO and not MMIO here? Just curious if this is the expected way to go.

Reply:
PIO is how NEMU virt does it. I copied that as it is a known working solution.

Reply:
Fair enough :)